### PR TITLE
Add test for DSR

### PR DESCRIPTION
**Pull request by epheph**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -130,6 +130,12 @@
 				}
 			}
 		},
+		"@zoltu/typescript-transformer-append-js-extension": {
```
Comment (MicahZoltu) [inline package-lock.json]: I think you need to `npm install` again, or perhaps your working tree is dirty.  This package and `ttypescript` are both added in my branch, but not here.  From this and some other comments you recently made in chat, I _think_ you somehow have a dirty checkout.

File: package.json [hunk 2]
```diff
@@ -17,17 +17,21 @@
...
 		"node-fetch": "2.6.0",
 		"solc": "0.5.12",
 		"ts-node": "8.4.1",
-		"typescript": "3.7.2"
+		"typescript": "3.7.2",
+		"ttypescript": "1.5.8"
+	},
+	"dependencies": {
 	},
-	"dependencies": {},
```
Comment (MicahZoltu) [inline package.json]: These changes look like they are from my branch.  Recommend cleaning them out of here.

File: package.json [hunk 3]
```diff
@@ -17,17 +17,21 @@
...
 	"scripts": {
+		"e": "ts-node -v && ts-node -v",
```
Comment (MicahZoltu) [inline package.json]: What does this do?

File: test/helpers.ts [hunk 4]
```diff
@@ -59,3 +59,7 @@ export function getGanacheControls(rpc: JsonRpc): GanacheControls {
 		},
 	}
 }
+
+export async function mineBlock(actor: Actor) {
+	return actor.rpc.sendEth(actor.signer.address, 0n)
+}
```
Comment (MicahZoltu) [inline test/helpers.ts]: Hax.  I wonder if we should name this `triggerInstasealBlock` to help distinguish it from the other `mine` function in here.

File: test/test.ts [hunk 5]
```diff
@@ -64,6 +64,46 @@ describe('DaiHrd', () => {
...
+		await alice.dai.approve(alice.daiHrd.address, MAX_APPROVAL)
+		const chiBeforeDeposit = await alice.daiHrd.calculatedChi_()
+		await ganache.advanceTime(2)
+
+		await alice.daiHrd.deposit(attodaiToDeposit)
+		await ganache.advanceTime(2)
+
+		await mineBlock(alice)
+		const chiAfterDeposit = await alice.daiHrd.calculatedChi_()
+		await ganache.advanceTime(14)
```
Comment (MicahZoltu) [inline test/test.ts]: If we are using `ganache.advanceTime(...)`, it means that this test won't work in Parity.  This is fine, but in that case why not use `ganache.mine`?  It feels odd to do `ganache.advanceTime(...)` followed by a general purpose `mine`?

File: test/helpers.ts (hunk 4 above)
Comment (epheph) [reply test/helpers.ts]: much better name, since we aren't mining at all

File: test/test.ts (hunk 5 above)
Comment (epheph) [reply test/test.ts]: well, advancing time and triggering a transaction are two different things, right?

Comment (MicahZoltu) [reply test/test.ts]: The one two lines up does:
```typescript
await ganache.advanceTime(2)
await mineBlock(alice)
```
Why not just do this?
```typescript
await ganache.mine(now + 2)
```

I suppose there is value in consistency and using separate methods for changing time and mining.

**Pull request merged**